src/gui/executionlogwidget.cpp

```diff
@@ -32,75 +32,46 @@
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_msgFilterModel(new LogFilterModel(types, this))
 {
     m_ui->setupUi(this);
 
+    LogModel *msgModel = new LogModel(this);
+    m_msgFilterModel->setSourceModel(msgModel);
+    LogListView *msgList = new LogListView(this);
+    msgList->setModel(m_msgFilterModel);
+    connect(msgList, &LogListView::resetModel, [msgModel]() { msgModel->reset(); });
```

Comment:
Seems a good idea to me to turn msgModel->reset() into a public slot and you can invoke it without the lambda.

Reply:
It's incorrect usage of signals. Signals should be used to inform about some events, but not to encourage something external to take some action.

Reply:
My first thought was ->

void LogListView::resetModel() {
  auto model = qobject_cast<BaseLogModel *>(model());
  model->resetModel();
}
but this will not work, because we have a filter mode inside MessageView, will have to add another check from FilterModel and then cast its model to baselogmodel, doesn't it will spill the implementation detail everywhere, any suggestion?

Reply:
I moved the contextmenuhandling in executionlogwidget please have a look..